ambry-commons/src/main/java/com/github/ambry/commons/NettySslFactory.java

```diff
@@ -73,10 +73,12 @@ public SSLEngine createSSLEngine(String peerHost, int peerPort, Mode mode) {
       sslParams.setEndpointIdentificationAlgorithm(endpointIdentification);
       sslEngine.setSSLParameters(sslParams);
     }
-    String[] supportedProtocols = sslEngine.getSupportedProtocols();
-    String[] enabledProtocols = sslEngine.getEnabledProtocols();
-    logger.info("NettySslFactory creating SSLEngine for {}: supportedProtocols: {}, enabledProtocols: {}", mode.name(),
-        String.join(",", supportedProtocols), String.join(",", enabledProtocols));
+    if (logger.isTraceEnabled()) {
+      String[] supportedProtocols = sslEngine.getSupportedProtocols();
+      String[] enabledProtocols = sslEngine.getEnabledProtocols();
+      logger.trace("NettySslFactory creating SSLEngine for {}: supportedProtocols: {}, enabledProtocols: {}", mode.name(),
+          String.join(",", supportedProtocols), String.join(",", enabledProtocols));
```

Comment:
How many times does NettySslEngine get created ? once or more ?

Reply:
quite a lot. Every time we received a new connection from client, we create it. Every time we created a connection to ambry-server, we create it.